sign_api/src/rpc/params/session_settle.rs

```diff
@@ -0,0 +1,57 @@
+//! https://specs.walletconnect.com/2.0/specs/clients/sign/rpc-methods
+//! #wc_sessionsettle
+
+use serde::{Deserialize, Serialize};
+
+use super::{IrnMetadata, Metadata, Relay};
+
+pub(super) const IRN_REQUEST_METADATA: IrnMetadata = IrnMetadata {
+    tag: 1102,
+    ttl: 300,
+    prompt: false,
+};
+
+pub(super) const IRN_RESPONSE_METADATA: IrnMetadata = IrnMetadata {
+    tag: 1103,
+    ttl: 300,
+    prompt: false,
+};
+
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, Clone, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Controller {
+    pub public_key: String,
+    pub metadata: Metadata,
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, Clone, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct SettleNamespaces {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub eip155: Option<SettleNamespace>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub cosmos: Option<SettleNamespace>,
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, Clone, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct SettleNamespace {
+    pub accounts: Vec<String>,
+    pub methods: Vec<String>,
+    pub events: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub extensions: Option<Vec<Self>>,
+}
+
+#[derive(Debug, Serialize, PartialEq, Eq, Deserialize, Clone, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct SessionSettleRequest {
+    pub relay: Relay,
+    pub controller: Controller,
+    pub namespaces: SettleNamespaces,
+    /// uSecs contrary to what documentation says (secs).
+    pub expiry: u64,
```

Comment:
I've requested more info from the team on this -- if microseconds are used indeed, we need to update the documentation.

Reply:
When I bump down the expiry to 300 (5 min) in seconds, then I get an expiry error on `session_settle`:
![image](https://github.com/WalletConnect/WalletConnectRust/assets/103181555/0b51fcf3-37a2-4d87-b6e2-3c837a10c1b7)

Reply:
When I have poked around it a long time ago, I used example wallet and put some debug statements to see the raw data passed around, and this was the example payload for `session_settle`:
![image](https://github.com/WalletConnect/WalletConnectRust/assets/103181555/d37aa8c8-5fad-4e11-bdac-d05ee882c825)

Reply:
Which makes is almost 30min in usec.

Reply:
Thank you for extra info -- will make sure I follow this up with API team to understand whether the issue is still there.

Reply:
Hey @silvestrst-crypto, chiming in from JS team. That `session` expiry is indeed in `milliseconds` while all? other `TTL`/user-defined expirires are in seconds. I will get the team to update the docs, thank you 💯

Reply:
Are you guys sure its not `uSec`? I have just changed `uSec` to `ms` in my session example:
```
fn create_settle_request(responder_public_key: String) -> RequestParams {
    RequestParams::SessionSettle(SessionSettleRequest {
        relay: Relay {
            protocol: SUPPORTED_PROTOCOL.to_string(),
            data: None,
        },
        controller: Controller {
            public_key: responder_public_key.to_string(),
            metadata: Metadata {
                name: format!("Rust session example: {}", Utc::now()),
                icons: vec!["https://www.rust-lang.org/static/images/rust-logo-blk.svg".to_string()],
                ..Default::default()
            },
        },
        namespaces: supported_settle_namespaces(supported_account()),
        expiry: 300000, // 5 min in ms
    })
}
```

And it is expiring still with the same message as in the screenshot above.

Reply:
Worked, must be the same concept as for the `wc_sessionrequest`, just wasn't explicitly mentioned in the docs.